.github/workflows/pre_commit_checks.yml

```
@@ -0,0 +1,215 @@
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      # ─────────────────────────────────────────────
+      # 1. BRANCH NAMING CONVENTION
+      # ─────────────────────────────────────────────
+      - name: Validate branch naming convention
+        if: github.event_name == 'pull_request'
+        run: |
```

actionlint (1.7.11)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)